src/data_types/rev_reg.rs

```diff
@@ -35,22 +37,146 @@ pub struct RevocationRegistryDelta {
 
 impl Validatable for RevocationRegistryDelta {}
 
+#[derive(Clone, Copy, Debug, Serialize)]
+pub struct UrsaRevocationRegistry(Accumulator);
+
+impl TryFrom<&str> for UrsaRevocationRegistry {
+    type Error = Error;
+
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        let accum = Accumulator::from_string(value)?;
+        Ok(UrsaRevocationRegistry(accum))
+    }
+}
+
+impl From<ursa::cl::RevocationRegistry> for UrsaRevocationRegistry {
+    fn from(value: ursa::cl::RevocationRegistry) -> Self {
+        let s = serde_json::to_string(&value).unwrap();
+        serde_json::from_str(&s).unwrap()
+    }
+}
+
+impl From<UrsaRevocationRegistry> for ursa::cl::RevocationRegistry {
+    fn from(value: UrsaRevocationRegistry) -> Self {
+        let s = serde_json::to_string(&value).unwrap();
+        let json = format!("{{\"accum\": {s}}}");
+        serde_json::from_str(&json).unwrap()
+    }
+}
+
+// impl TryFrom<ursa::cl::RevocationRegistry> for UrsaRevocationRegistry {
+//     type Error = Error;
+//
+//     fn try_from(value: ursa::cl::RevocationRegistry) -> Result<Self, Self::Error> {
+//         let s = serde_json::to_string(&value)?;
+//         Ok(serde_json::from_str(&s)?)
+//     }
+// }
+//
+// impl TryFrom<UrsaRevocationRegistry> for ursa::cl::RevocationRegistry{
+//     type Error = Error;
+//
+//     fn try_from(value: UrsaRevocationRegistry) -> Result<Self, Self::Error> {
+//         let s = serde_json::to_string(&value)?;
+//         Ok(serde_json::from_str(&s)?)
+//     }
+// }
+
+// impl Serialize for UrsaRevocationRegistry {
+//     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+//     where
+//         S: Serializer,
+//     {
+//         let mut state = serializer.serialize_tuple_struct("UrsaRevocationRegistry", 1)?;
+//         state.serialize_field(&self.0)?;
+//         state.end()
+//     }
+// }
+
+impl<'de> Deserialize<'de> for UrsaRevocationRegistry {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct UrsaRevocationRegistryVisitor;
+
+        impl<'de> Visitor<'de> for UrsaRevocationRegistryVisitor {
+            type Value = UrsaRevocationRegistry;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                write!(formatter, "string or map")
+            }
+
+            fn visit_str<E: serde::de::Error>(
+                self,
+                value: &str,
+            ) -> Result<UrsaRevocationRegistry, E> {
+                let accum = Accumulator::from_string(value).map_err(de::Error::custom)?;
+                Ok(UrsaRevocationRegistry(accum))
+            }
+
+            // fn visit_seq<V>(self, mut seq: V) -> Result<UrsaRevocationRegistry, V::Error>
+            // where
```

Comment:
Can this he removed?

Reply:
Yeah I created a PR, but it should be more a draft as not all the tests are working, yet. Once all the tests are working I will remove the unused code.